src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCacheExpiry.java

```diff
@@ -35,7 +35,7 @@ public long expireAfterUpdate(
         long currentTime,
         @NonNegative long currentDuration
     ) {
-        return currentDuration;
+        return TimeUnit.SECONDS.toNanos(cachedIdamCredential.expiresIn - refreshTokenBeforeExpiry);
```

Comment:
as we do not refresh the token if this works it will change the timer for existing token which is wrong.
for example remaining time is 300 for current token if this works it will again make it 28800.

Reply:
this is for replacing the token

Reply:
where do you replace the token ?

Reply:
I guess we should either have correct implementation of this method or always throw exception.

Reply:
I think returning `currentDuration` is not wrong. if some thing is triggering this method is not us.we are not refreshing the token. if you think returning currentDuration is wrong , throwing exception might a better option.